### ISSUE TITLE
Control Panel Parts List for 240V/50A and 120V/15A components

### ISSUE THREAD

**Issue by shawn0x1** [FinalGravityBrewing/control_panel#1]

**Comment by shawn0x1** [comment]
@GOFmugsy, do you mind talking to the guys at work to see about feasibility of using 4-5500W heating elements in the brew system?  Idea is to be able to brew back to back batches and heat water/wort faster.  I think main concern, like we discussed last night, is what kind of residential requirements at the panel we will need. 

We are basing this feasibility off of [the >1BBL setup from the Electric Brewery](http://www.theelectricbrewery.com/forum/viewtopic.php?t=24918) and the following real world examples from the thread referenced above:
> * Heat 30 gallons of strike water from 70F to 155F: 68 minutes with 5500W, 42 minutes with 9000W, 34 minutes with 11000W
> * Bring 24 gallons of wort from 150F to boil: 40 minutes with 5500W, 24 minutes with 9000W, 20 minutes with 11000W

**Comment by shawn0x1** [comment]
@skeptonomicon and I did some brainstorming today and I think with 2 5500W heaters we can do .79C/min change with 40 gallons.  This takes 40 gal from 24C-100C in 1hr w/ 11000W.  This requires a 50A breaker which is the size of a stove.  4-5500W heaters at 240V requires around a 92A breaker which will be very hard for anyone in a residential setting to have installed.  So, maybe best way to go is to have 4-5500W elements, but only allow 2 to be on at a time.

**Comment by shawn0x1** [comment]
@ryho @rlmcpherson @GOFmugsy, let me know what you guys think...

**Comment by GOFmugsy** [comment]
I don't work in distribution so the neighborhood-level stuff is mostly speculation on my and my coworker's parts. But I'm thinking it would be good to ask the club to check on the sizing of their master-breakers and the sizes of any 240 circuits they may already have. It'll help us determine what sizing we could go with and possible hosts. 

From my understanding the master-breaker sizing is agreed upon with Georgia Power and the sizing of their street transformer. That's really going to be our limiting factor I think. Thoughts?

**Comment by shawn0x1** [comment]
@GOFmugsy, I think if use a safe start mentioned in issue #4 to limit the panel to 2 elements used at a time we can get away with a 240V 60A breaker which I personally have room for in my service panel and I assume would be available to most people? Anything less would result in a long brew day, IMO.

**Comment by shawn0x1** [comment]
Like I said at the last Final Gravity meeting, 

I am up for driving initial design concept (with more input, please), fabrication of the control panel box (hole punches, cut outs, metal work, etching/labeling, etc), but I will depend on the rest of you to do the wiring, cable fab, and any coding we need.  I don't mind putting the time in on the front end to do design a panel that creates a good brew day experience, by saving on time heating the water and boiling, etc.  However, I think we will need a time commitment from each of you and now is the time to speak up if you think you cannot help out.  Participation has been rather "thin" so far.  I am ready to order at least the 240V parts with the assumption that we may use some a single-board computer (RaspPi or Arduino) to implement a safety start with 5V control signals going through the switches to relays and the computer of choice.  I don't think that it will be too much coding, but feel free to speak up if whoever plans on helping on that end thinks it will be too much work.  We will need a power supply for whatever single-board computer we pick, but all I think we want it to do is control two elements being on at a time and possibly sparge level, which is 5 i/o.  Let me know what you guys think ASAP, other options would be to go with a turnkey solution which will most definitely save time up front, but will be longer brew days, or we can abandon this project and just do something else with the money.

**Comment by wgr6** [comment]
Way late to the party here... sorry.  I really want to help out with this project.  Unfortunately, my kids don't leave me a whole lot of time.  Realistically the only time I have is 8:00-10:00 pm on weekdays, and 8:00-12:00 on weekends.  Occasionally, maybe once every two weeks, I could get a chunk of 4-5 hours if we all get together somewhere to work on things.  So, I'll say I can commit 5 hours/week on average.

Right now I think I'm most confused on the timeline for this project.  If we want a finished project 3 months from now, I think we could/should do this ourselves.  If we want it ASAP, let's just buy a finished product... do we have the money for that?  Could we just "tax" group brews to pay for a pricy system we can't immediately afford?  Like $5 per person until it's "paid off". Just throwing out random thoughts.  My honest feeling is we should just discuss all of this again face to face, either at the next group meeting or something we set up.  

Finally, have we completely eliminated the possibility of a natural gas system?  Would most households just not have the pressure we'd need to do a batch this big?  Also, are we shooting for the ability to do two stand brews?  It seems like a 1BBL system is already stretching the capabilities of a household.

**Comment by ryho** [comment]
Sorry about the lack of input, I'm not great about checking my GitHub notifications. Usually it's just people pestering me.

I like the idea of the 5V system, and seems like all the parts will be cheaper that way as well.
BTW the BrewTroller project is basically abandonware at this point. Half the stuff on their site is sold out. The last commit to the project was in 2014. Someone on HombrewTalk got it working on an Arduino Mega though, and they've got a fork that was last updated in 2016. That won't be a problem for a little club project I suppose. If it works, it works.
I've looked through the manuals, and opened their code with the Arduino IDE. The BrewTroller software is all in C, not Python, though that's not a problem. So, like you said it doesn't support dual elements for the mash tun and kettle like we want (there is a ticket for it though: https://github.com/BrewTroller/BrewTroller-Official/issues/50). It would be possible to add it. It's not going to be as simple as a few if statements, but I could do it. I'll have to add extra pin outs for that and plumb them through the code. I believe that that the BrewTroller board has extras pins on it. The Arduino Mega will since it has a lot more than the BrewTroller.

So the way the user would actually tell the BrewTroller system to turn 3 elements on at the same time is by having it set to boil while not being near boil temp (so that the system wants both boil coils on) and have the HLT set to hold temp, right? In that scenario we'd need to not actually allow that, and to show something on the little display telling the user that there's a problem.

So what's the pros and cons of the BrewTroller Pheonix and the Arduino mega? The former is 12V, and the Arduino is 5V. It might be trickier to draw up an equivalent parts list with all 5V parts, where as with the BrewTroller you can just buy exactly what they recommend. The BrewTroller also has ethernet so that you can control the system from a webpage. The UI on the BrewTroller does not seem that great, so the web UI could be pretty useful. I could even imagine hooking up an old iPad to this thing and controlling the temps and the settings through that. they make Ethernet adapters for iPads.

**Comment by shawn0x1** [comment]
I think the pros of the Pheonix are that it has mounted screw terminals and maybe more I/O?  IMO I think we should go with a Raspberry Pi, it may be more modular as people are using Python more and more these days.  I am pretty familiar with the brewtroller code, and yes, it has been pretty much abandoned.  Let's all talk about this at the meeting next Thursday.  Can @GOFmugsy @wgr6 @rlmcpherson and @skeptonomicon attend?  I saw that you will be there @ryho.  I got a good start on the parts list.  I need to look it over and add more to it, but [here](https://docs.google.com/spreadsheets/d/1ZKNQDeCOlhCeO5iMgSwde4qGQgvCIorXXLICjAbbJRo/edit?usp=sharing) is the list so far.

**Comment by wgr6** [comment]
I have some friends visiting.  They land at ATL at 8:00 Thursday. I could
potentially come until then... But that's not a lot of time.  Are people
getting dinner beforehand?

Wade

**Comment by GOFmugsy** [comment]
I will be at the next meeting, and apologies for any lack of participation on my part, just not sure what input is required at this point. I mean, @shawn0x1 , you have the most experience with this stuff, with only two elements at a time, this should be entirely doable. If you have a design that will work, let's order some shit and do it.

Give me a box and parts and I'll get the damn thing wired up. As far as design input though, I have no experience building a brewstand and can't provide much input. I will say, RPi would make me much happier than web/iPad...

**Comment by shawn0x1** [comment]
@GOFmugsy, no worries.  I think after talking more with @skeptonomicon we are going to go with the BCS-482.  it will save a ton in terms of buying separate PIDs and has ladder logic and a web interface and will do everything we need.  I opened issue #6 to hash out any questions on that solution.  @ryho, I saw you were posting on the brewtroller users group.  That is awesome, they need some life in that project.  I have a brewtroller on my personal stand if you want to take a look at it sometime.

**Comment by ryho** [comment]
Yeah, I was checking things out in case we go the BrewTroller route. Whatever code I write, I'd prefer for it to get merged into the code base of the product we're using.  That way the club doesn't have to depend on me (or someone else) to maintain the code I write, but the future developers of that product will have to make sure that new versions work with my feature. I'm not planning on leaving the club or anything, but future proofing is an important part of developing something.

Will the BCS-482 do what we need for controlling 4 burners with only two on? Are there any demos of the web site or videos of the product or anything? I couldn't get a good feel for it from the documentation. Also, it looks like the device will only work if it is connected to the internet, so we would have to have an ethernet cable where ever we're using the system. The BrewTroller has a web app to control it, but it's also got the little display and the knob to fall back on for use off line. I'd hate to see us not be able to brew because the internet is down, but maybe there's a way to use the BCS offline.

We should make a doc of some sort to discuss the pros and cons of the different controller systems.

**Comment by skeptonomicon** [comment]
I think the device will need a network, but I don't think that it requires
the network to be connected to the internet. It might require a router just
so your control computer can connect to the BCS over ethernet.

**Comment by skeptonomicon** [comment]
But I agree we need to look very closely to the BCS before we commit. Other
issues like this might arise. I am also concerned with the lack of
documentation and the newness might imply there might be some bugs to work
through.

**Comment by skeptonomicon** [comment]
I was going to do this myself, but it looks like someone beet me to it.
https://forum.embeddedcc.com/viewtopic.php?f=13&t=2943

**Comment by rlmcpherson** [comment]
Just catching up on the conversation. I'll be at the meeting on Thursday. Talk to you then.
